src/gbinder_writer.c

```diff
@@ -697,6 +697,30 @@ gbinder_writer_append_remote_object(
     }
 }
 
+void
+gbinder_writer_append_byte_array(
+    GBinderWriter *self,
+    const void *byte_array,
+    gint32 len) /* since 1.0.12 */
+{
+    GBinderWriterData *data = gbinder_writer_data(self);
+
+    GASSERT(len >= 0);
+
+    if (G_LIKELY(data)) {
+        void* ptr;
+
+        g_byte_array_set_size(data->bytes, data->bytes->len + sizeof(len) + len);
+        ptr = (void*)(data->bytes->data + (data->bytes->len - sizeof(len) - len));
+        if (len > 0 && byte_array) {
+            *((gint32 *)ptr) = len;
+            ptr += sizeof(len);
+            memcpy(ptr, byte_array, len);
+        } else
```

Comment:
Brackets around the `else` block, please... And again, it's ok to leave it as is, but don't be surprised if I add brackets in a separate commit.